Build warnings [-Wpointer-sign] with LLVM/icx (tests/unit/bluetooth/at)

**_Reported by rahul kumar:_**

Test case with build warning raised from LLVM/icx: tests/unit/bluetooth/at

```
/home/vikram/Project/zephyr/subsys/bluetooth/host/at.c:203:14: warning:
      passing 'unsigned char *' to parameter of type 'const char *' converts
      between pointers to integer types with different sign [-Wpointer-sign]
        if (strncmp(at->buf, "CME ERROR", 9) == 0) {
                    ^~~~~~~
/home/vikram/Project/zephyr/lib/libc/minimal/include/string.h:26:35: note:
      passing argument to parameter 's1' here
extern int    strncmp(const char *s1, const char *s2, size_t n);
                                  ^
/home/vikram/Project/zephyr/subsys/bluetooth/host/at.c:233:14: warning:
      passing 'unsigned char *' to parameter of type 'const char *' converts
      between pointers to integer types with different sign [-Wpointer-sign]
        if (strncmp(at->buf, "RING", 4) == 0) {
                    ^~~~~~~
/home/vikram/Project/zephyr/lib/libc/minimal/include/string.h:26:35: note:
      passing argument to parameter 's1' here
extern int    strncmp(const char *s1, const char *s2, size_t n);
                                  ^
/home/vikram/Project/zephyr/subsys/bluetooth/host/at.c:250:22: warning:
      passing 'unsigned char *' to parameter of type 'const char *' converts
      between pointers to integer types with different sign [-Wpointer-sign]
        if (at_parse_result(at->buf, buf, &result) == 0) {
                            ^~~~~~~
/home/vikram/Project/zephyr/subsys/bluetooth/host/at.c:73:40: note: passing
      argument to parameter 'str' here
static int at_parse_result(const char *str, struct net_buf *buf,
                                       ^
/home/vikram/Project/zephyr/subsys/bluetooth/host/at.c:338:22: warning:
      passing 'unsigned char *' to parameter of type 'const char *' converts
      between pointers to integer types with different sign [-Wpointer-sign]
        if (!str_has_prefix(at->buf, prefix)) {
                            ^~~~~~~
/home/vikram/Project/zephyr/subsys/bluetooth/host/at.c:64:40: note: passing
      argument to parameter 'str' here
static bool str_has_prefix(const char *str, const char *prefix)
                                       ^
4 warnings generated.
  CC      subsys/bluetooth/host/hfp_hf.o
/home/vikram/Project/zephyr/subsys/bluetooth/host/hfp_hf.c:82:18: warning:
      passing 'u8_t *' (aka 'unsigned char *') to parameter of type 'char *'
      converts between pointers to integer types with different sign
      [-Wpointer-sign]
        ret = vsnprintk(buf->data, (net_buf_tailroom(buf) - 1), format, vargs);
                        ^~~~~~~~~
/home/vikram/Project/zephyr/include/misc/printk.h:49:48: note: passing
      argument to parameter 'str' here
extern __printf_like(3, 0) int vsnprintk(char *str, size_t size,
                                               ^
/home/vikram/Project/zephyr/subsys/bluetooth/host/hfp_hf.c:373:16: warning:
      passing 'unsigned char *' to parameter of type 'const char *' converts
      between pointers to integer types with different sign [-Wpointer-sign]
                if (!strncmp(hf_at->buf, handlers[i].cmd,
                             ^~~~~~~~~~
/home/vikram/Project/zephyr/lib/libc/minimal/include/string.h:26:35: note:
      passing argument to parameter 's1' here
extern int    strncmp(const char *s1, const char *s2, size_t n);
                                  ^
/home/vikram/Project/zephyr/subsys/bluetooth/host/hfp_hf.c:657:14: warning:
      assigning to 'unsigned char *' from 'char [140]' converts between pointers
      to integer types with different sign [-Wpointer-sign]
                hf->at.buf = hf->hf_buffer;
                           ^ ~~~~~~~~~~~~~
3 warnings generated.
/home/vikram/Project/zephyr/tests/unit/bluetooth/at/src/main.c:53:9: warning:
      assigning to 'unsigned char *' from 'char [140]' converts between pointers
      to integer types with different sign [-Wpointer-sign]
        at.buf = buffer;
               ^ ~~~~~~
1 warning generated.

```

How to setup llvm/icx
# update the icx compiler with 0214 build from ISSM team
#  source <path_to_llvm>/linux_prod/linux/bin/xcompilervars_arch.sh ia32 mcu
#  cd zephyr
#  source zephyr-env.sh
#  export ISSM_INSTALLATION_PATH=<path_to_issm_toolchain>/issm_2016.1.057
#  export ZEPHYR_GCC_VARIANT=issm

How-to-reproduce:
# cd tests/unit/bluetooth/at
# make pristine
#  make BOARD=quark_se_c1000_devboard CC=icx

Expected result : Build should pass without any warning or error.

Actual result : Observed build warnings.

Attached captured compilation log.

(Imported from Jira ZEP-2161)

--- a/subsys/bluetooth/host/at.c
+++ b/subsys/bluetooth/host/at.c
@@ -92,7 +92,7 @@ static int get_cmd_value(struct at_client *at, struct net_buf *buf,
 {
 	int cmd_len = 0;
 	u8_t pos = at->pos;
-	const unsigned char *str = buf->data;
+	const char *str = (char *)buf->data;
 
 	while (cmd_len < buf->len && at->pos != at->buf_max_len) {
 		if (*str != stop_byte) {
@@ -122,7 +122,7 @@ static int get_response_string(struct at_client *at, struct net_buf *buf,
 {
 	int cmd_len = 0;
 	u8_t pos = at->pos;
-	const unsigned char *str = buf->data;
+	const char *str = (char *)buf->data;
 
 	while (cmd_len < buf->len && at->pos != at->buf_max_len) {
 		if (*str != stop_byte) {

--- a/subsys/bluetooth/host/at.h
+++ b/subsys/bluetooth/host/at.h
@@ -89,7 +89,7 @@ typedef int (*handle_cmd_input_t)(struct at_client *at, struct net_buf *buf,
 				  enum at_cmd_type type);
 
 struct at_client {
-	unsigned char *buf;
+	char *buf;
 	u8_t pos;
 	u8_t buf_max_len;
 	u8_t state;

